### ISSUE TITLE
Version 1.11.x has exception not caught in CharacterReader.bufferUp

### ISSUE THREAD

**Issue by eos1d3** [jhy/jsoup#1005]
Exception is not caught and not possible to catch in `CharacterReader.bufferUp`

Tested with version 1.11.2,  with both

`Doc doc = Jsoup(....)`

or 

```
Response reponse = Jsoup(...)
Document doc = response.parse();
```

The method tested with both `throws IOException, UncheckedIOException, SocketException` and also `try ... catch (Exception e)`

None of them can catch the exception. But using 1.10.3 will never find the problem.

This is the error output:

```
Exception in thread "Thread-8" org.jsoup.UncheckedIOException: java.net.SocketException: Connection reset

at org.jsoup.parser.CharacterReader.bufferUp(CharacterReader.java:60)
at org.jsoup.parser.CharacterReader.Current(CharacterReader.java:85)
at org.jsoup.parser.TokeniserState.readData(TokeniserState.java:1669
at org.jsoup.parser.TokeniserState.access$200(TokeniserState.java:8)
at org.jsoup.parser.TokeniserState$6.read(TokeniserState.java:76)
at org.jsoup.parser.Tokeniser.read(Tokeniser.java 45)
at org.jsoup.parser.TreeBuilder.runParser(TreeBuilder.java:51)
at org.jsoup.parser.TreeBuilder.parse(TreeBuilder.java:45)
at org.jsoup.parser.Parser.parseInput(Parser.java:39)
at org.jsoup.helper.DataUtil.parseInputStream(DataUtil.java:151)
at org.jsoup.helper.HttpConnection$Response.parse(HttpConnection.java:832)
at MyApp.mediasWeb(Instagrammjava:280)
at MyApp.Mediascanner.run(Mediascanner.java:31)

```

**Comment by mvhgroup** [comment]
Seeing that a lot as well. That stream reader was added in 1.11.1, so you wont see the error, but the issue then is the out of memory error, which the stream reader fixes.

**Comment by dentex** [comment]
@jhy is this the same stuff as #980 ? You may want to close also this one.

**Comment by jhy** [comment]
Thanks, yes, this is the same issue/fix as #980
